pkg/apis/azuredisk/v1alpha1/types.go

```diff
@@ -87,26 +85,59 @@ type AzVolumeAttachmentList struct {
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="NodeName",type=string,JSONPath=`.spec.nodeName`,description="Name of the Node which this AzDriverNode object represents."
+// +kubebuilder:printcolumn:name="ReadyForVolumeAllocation",type=boolean,JSONPath=`.status.readyForVolumeAllocation`,description="Indicates if the azure persistent volume driver is ready for new pods which use azure persistent volumes."
+// +kubebuilder:printcolumn:name="LastHeatbeatTime",type=integer,JSONPath=`.status.lastHeatbeatTime`,description="Represents the time stamp at which azure persistent volume driver sent a heatbeat."
 type AzDriverNode struct {
 	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
-
-	Spec   AzDriverNodeSpec   `json:"spec"`
-	Status AzDriverNodeStatus `json:"status"`
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// spec defines the desired state of a AzDriverNode.
+	// Required.
+	Spec   AzDriverNodeSpec   `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+
+	// status represents the current state of AzDriverNode.
+	// If this is nil or empty, clients should prefer other nodes
+	// for persistent volume allocations or pod places for pods which use azure persistent volumes.
+	// +optional
+	Status *AzDriverNodeStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
 }
 
-// AzDriverNodeSpec is the spec for a AzDriverNode resource
+// AzDriverNodeSpec is the spec for a AzDriverNode resource.
 type AzDriverNodeSpec struct {
-	UnderlyingNode string `json:"underlyingNode"`
+	// Name of the node which this AzDriverNode represents.
+	// Required.
+	NodeName *string `json:"nodeName" protobuf:"bytes,1,opt,name=nodeName"`
 }
 
-// AzDriverNodeStatus is the status for a AzDriverNode resource
+// AzDriverNodeStatus is the status for a AzDriverNode resource.
 type AzDriverNodeStatus struct {
-	UnderlyingNode int32 `json:"underlyingNode"`
+	// LastHeatbeatTime represents the timestamp when a heatbeat was sent by driver node plugin.
+	// A recent timestamp means that node-plugin is responsive and is communicating to API server.
+	// Clients should not solely reply on LastHeatbeatTime to ascertain node plugin's health state.
+	// +optional
+	LastHeatbeatTime *int64 `json:"lastHeatbeatTime,omitempty" protobuf:"varint,1,opt,name=lastHeatbeatTime"`
+
+	// HealthStatus is the health message posted by node plugin along with the heartbeat
+	// This field is for status\debugging purpose only.
+	// Clients should not use\parse this field to make decisions in code.
+	// +optional
+	HealthStatus *string `json:"healthStatus,omitempty" protobuf:"bytes,2,opt,name=healthStatus"`
```

Comment:
I'd think that a status that could be consumed programatically would be useful. Perhaps use a `HealthStatus` field which can be consumed programatically and a `HealthDescription` field which is for human consumption?

Reply:
I plan to actually use this. This is a common practice apparently across upstream and I think I like it.
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties

Reply:
take a look at my change and the comment and let me know if you think we should change this.